server/src/Api/Controllers/FlashcardsController.cs

```diff
@@ -53,4 +55,29 @@ public IActionResult DeleteFlashcard(int flashcardId)
         var isSuccess = _flashcardService.DeleteFlashcard(flashcardId);
         return isSuccess ? Ok() : NotFound();
     }
+
+    [HttpPost]
+    [Route("import-csv")]
+    [Consumes("multipart/form-data")]
+    public IActionResult ImportFlashcardsFromCsv(IFormFile file)
```

Comment:
Controllers should be lightweight, they should not handle exceptions, etc. You can create custom exceptions and map them (look into DependencyInjection in Api project for exception mapping)

Reply:
In progress